stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt

```diff
@@ -680,10 +683,33 @@ internal class ChatDomainImpl internal constructor(
     internal suspend fun retryFailedEntities() {
         delay(1000)
         // retry channels, messages and reactions in that order..
-        val channelEntities = repos.channels.retryChannels()
+        val channels = retryChannels()
         val messages = retryMessages()
         val reactions = retryReactions()
-        logger.logI("Retried ${channelEntities.size} channel entities, ${messages.size} messages and ${reactions.size} reaction entities")
+        logger.logI("Retried ${channels.size} channel entities, ${messages.size} messages and ${reactions.size} reaction entities")
+    }
+
+    @VisibleForTesting
+    internal suspend fun retryChannels(): List<Channel> {
+        return repos.selectChannelsSyncNeeded().onEach { channel ->
+            val result = client.createChannel(
+                channel.type,
+                channel.id,
+                channel.members.map(UserEntity::getUserId),
+                channel.extraData
+            ).await()
+
+            when {
+                result.isSuccess -> {
+                    channel.syncStatus = SyncStatus.COMPLETED
+                    repos.insertChannel(channel)
+                }
+                result.isError && result.error().isPermanent() -> {
```

Comment:
What happens if the error is not permanent?

Reply:
Feels like don't update right now these failed channels. I'm not an author of this logic. Just copied as it was [before](https://github.com/GetStream/stream-chat-android/blob/033455e584b93b6adadbce03e004df9256b68d9e/stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/repository/ChannelRepository.kt#L78)